### PR TITLE
PHPCS: JSON Endpoints 3 of ?

### PR DESCRIPTION
**Pull request by sdixon194**
Fixes the remainder of #21959

#### Changes proposed in this Pull Request:

* PHPCS changes for the remaining json endpoint files in that issue.

#### Jetpack product discussion

n/a
#### Does this pull request change what data or activity we track or use?

no
#### Testing instructions:

* Proof/test?
'
*

### CODE REVIEW

File: projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-themes-install-endpoint.php [hunk 1]
```diff
@@ -147,25 +174,39 @@ protected function validate_themes() {
...
 	protected static function is_installed_theme( $theme ) {
 		$wp_theme = wp_get_theme( $theme );
 		return $wp_theme->exists();
 	}
 
+	/**
+	 * Download the wpcom theme.
+	 *
+	 * @param string $theme - the theme to download.
+	 */
```
Comment (anomiex) [inline projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-themes-install-endpoint.php:193]: `@return`?

File: projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-themes-modify-endpoint.php [hunk 2]
```diff
@@ -83,21 +125,24 @@ function update() {
 		return true;
 	}
 
-	function update_translations() {
+	/**
+	 * Update translations.
+	 */
```
Comment (anomiex) [inline projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-themes-modify-endpoint.php:130]: `@return`?

File: projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-updates-status-endpoint.php [hunk 3]
```diff
@@ -1,23 +1,35 @@
...
+	/**
+	 * Needed capabilities.
+	 *
+	 * @var string
+	 */
 	protected $needed_capabilities = 'manage_options';
 
+	/**
+	 * Endpoint callback.
+	 */
```
Comment (anomiex) [inline projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-updates-status-endpoint.php:18]: `@return`?

File: projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-user-connect-endpoint.php [hunk 4]
```diff
@@ -1,21 +1,48 @@
...
+	 * The user token.
+	 *
+	 * @var string
+	 */
 	private $user_token;
 
-	function result() {
+	/**
+	 * The endpoint callback.
+	 */
```
Comment (anomiex) [inline projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-user-connect-endpoint.php:34]: `@return`?

File: projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-user-connect-endpoint.php [hunk 5]
```diff
@@ -1,21 +1,48 @@
...
 		( new Tokens() )->update_user_token( $this->user_id, sprintf( '%s.%d', $this->user_token, $this->user_id ), false );
 		return array( 'success' => ( new Connection_Manager( 'jetpack' ) )->is_user_connected( $this->user_id ) );
 	}
 
-	function validate_input( $user_id ) {
+	/**
+	 * Validate input.
+	 *
+	 * @param int $user_id - the User ID.
+	 */
```
Comment (anomiex) [inline projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-user-connect-endpoint.php:44]: `@return`?

File: projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-user-create-endpoint.php [hunk 6]
```diff
@@ -1,38 +1,66 @@
...
+	 * User data.
+	 *
+	 * @var array
+	 */
 	private $user_data;
 
-	function result() {
+	/**
+	 * Endpoint callback.
+	 */
```
Comment (anomiex) [inline projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-user-create-endpoint.php:26]: `@return`?

File: projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-user-create-endpoint.php [hunk 7]
```diff
@@ -1,38 +1,66 @@
...
 		return $this->create_or_get_user();
 	}
 
-	function validate_input( $object ) {
+	/**
+	 * Validate the input.
+	 *
+	 * @param object $object - the object.
+	 *
+	 * @return bool
```
Comment (anomiex) [inline projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-user-create-endpoint.php:36]:
```suggestion
	 * @return bool|WP_Error
```

File: projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-user-create-endpoint.php [hunk 8]
```diff
@@ -61,10 +89,17 @@ function create_or_get_user() {
 		return $this->get_user( $user->ID );
 	}
 
+	/**
+	 * Get the user.
+	 *
+	 * @param int $user_id - the user ID.
+	 *
+	 * @return object
```
Comment (anomiex) [inline projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-user-create-endpoint.php:97]: While it's technically redundant, I'd still do like
```suggestion
	 * @return object|WP_Error
```
to make it clear how errors are reported.

File: projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-user-create-endpoint.php [hunk 9]
```diff
@@ -1,38 +1,66 @@
...
 			return new WP_Error( 'input_error', __( 'user login is required', 'jetpack' ) );
 		}
 		return parent::validate_input( $object );
 	}
 
-	function create_or_get_user() {
+	/**
+	 * Create or get the user.
+	 *
+	 * @return object
```
Comment (anomiex) [inline projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-user-create-endpoint.php:56]:
```suggestion
	 * @return object|false
```

File: projects/plugins/jetpack/json-endpoints/jetpack/class.wpcom-json-api-get-option-endpoint.php [hunk 10]
```diff
@@ -11,18 +14,37 @@ class WPCOM_JSON_API_Get_Option_Endpoint extends Jetpack_JSON_API_Endpoint {
...
 		return array( 'option_value' => get_option( $this->option_name ) );
 	}
 
-	function validate_input( $object ) {
-		$query_args = $this->query_args();
+	/**
+	 * Validate the input.
+	 *
+	 * @param object $object - unused, for parent class compatability.
+	 */
```
Comment (anomiex) [inline projects/plugins/jetpack/json-endpoints/jetpack/class.wpcom-json-api-get-option-endpoint.php:45]: `@return`?

File: projects/plugins/jetpack/json-endpoints/jetpack/class.wpcom-json-api-update-option-endpoint.php [hunk 11]
```diff
@@ -12,8 +25,13 @@ function result() {
 		return parent::result();
 	}
 
-	function validate_input( $object ) {
-		$input = $this->input();
+	/**
+	 * Validate the input.
+	 *
+	 * @param object $object - the object we're validating.
+	 */
```
Comment (anomiex) [inline projects/plugins/jetpack/json-endpoints/jetpack/class.wpcom-json-api-update-option-endpoint.php:32]: `@return`?

File: projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-themes-install-endpoint.php [hunk 12]
```diff
@@ -58,30 +84,31 @@ protected function install() {
...
-			return  new WP_Error( 'install_error', $error, 400 );
+			return new WP_Error( 'install_error', $error, 400 );
 		}
 
 		return true;
 	}
 
+	/**
+	 * Validate the themes.
+	 */
```
[the remark this answers is not in the document]
Comment (kraftbj) [reply projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-themes-install-endpoint.php:106]: `@return` isn't required by our coding standards. I think they're nice to have (and when we have them, they should be right and accurate), but I'm not sure if we should require them.

[the remark this answers is not in the document]
Comment (anomiex) [reply projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-themes-install-endpoint.php:106]: Personally I'm a bit surprised it's not.

If I ever get around to trying to check things with Psalm, we might wind up having to document return values then.

**Comment by sdixon194** [comment]
Think I got all the return statements.

File: projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-themes-install-endpoint.php [hunk 13]
```diff
@@ -190,6 +192,8 @@ protected static function is_installed_theme( $theme ) {
 	 * Download the wpcom theme.
 	 *
 	 * @param string $theme - the theme to download.
+	 *
+	 * @return string
```
Comment (anomiex) [inline projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-themes-install-endpoint.php:196]:
```suggestion
	 * @return string|WP_Error
```

**Review by anomiex — APPROVED** [review]
Let's go for it!

**Pull request merged**